### PR TITLE
Added assertHubConnection

### PR DESCRIPTION
**Pull request by bamandel**

### CODE REVIEW

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/Application.groovy [hunk 1]
```diff
@@ -230,7 +231,7 @@ class Application {
                 logger.debug(e.getCause().getMessage(), e.getCause());
             }
             exitCodeType = e.getExitCodeType();
-        } else if (e instanceof IntegrationException) {
+        } else if (e instanceof IntegrationException || e instanceof IntegrationCertificateException) {
```
Comment (ekerwin) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/Application.groovy]: Please, oh please tell me that IntegrationCertificateException isn't an IntegrationException...

Comment (ekerwin) [reply src/main/groovy/com/blackducksoftware/integration/hub/detect/Application.groovy]: This actually should be IllegalStateException as the underlying code in HubServerConfigBuilder, line 72, can throw that.

Comment (bamandel) [reply src/main/groovy/com/blackducksoftware/integration/hub/detect/Application.groovy]: My original idea was to add IllegalStateException instead but then I figured many other processes could throw that. I wanted this specific handling when certificate is wrong or credentials were wrong, so I added the specific exception. Do you think it'd be a good idea to make IntegrationCertificateExcpetion an IntegrationException? I think that'd be the best option.

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/hub/HubServiceWrapper.groovy [hunk 2]
```diff
@@ -88,24 +88,13 @@ class HubServiceWrapper {
...
-                slf4jIntLogger = new SilentLogger();
+                assertHubConnection()
             }
-            HubServerConfig hubServerConfig = createHubServerConfig(slf4jIntLogger);
-
-            final RestConnection connection = hubServerConfig.createRestConnection(slf4jIntLogger)
-            connection.connect()
-            logger.info("Connection to the Hub was successful")
             return true;
-        } catch (IllegalStateException e) {
```
Comment (bamandel) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/hub/HubServiceWrapper.groovy]: I believe removing this and not modifying any other logs will be fine as the exception message will say that there's an issue with your cert. That's enough of a log imo

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/hub/HubServiceWrapper.groovy [hunk 3]
```diff
@@ -114,6 +103,23 @@ class HubServiceWrapper {
         return false;
     }
 
+    public void assertHubConnection() {
```
Comment (jamesrichard91) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/hub/HubServiceWrapper.groovy]: We should kill this method and just use `assertHubConnection(new SilentLogger()`

**Pull request merged**